napari/_vispy/layers/base.py

```diff
@@ -120,8 +122,32 @@ def _on_visible_change(self):
     def _on_opacity_change(self):
         self.node.opacity = self.layer.opacity
 
-    def _on_blending_change(self):
-        blending_kwargs = BLENDING_MODES[self.layer.blending]
+    def _on_blending_change(self, event=None):
+        blending = self.layer.blending
+        blending_kwargs = BLENDING_MODES[blending].copy()
+
+        if self.first_visible:
+            # if the first layer, then we should blend differently
+            # (ignore the canvas color but allow alpha blending)
+            if blending in ('minimum', 'additive'):
+                src_color_blending = 'one'
+                dst_color_blending = 'zero'
+            else:
+                src_color_blending = 'src_alpha'
+                dst_color_blending = 'one_minus_src_alpha'
+            blending_kwargs = dict(
+                depth_test=True,
```

Comment:
Makes sense to me :) For the depth test, I did it the way thinking that we never want to blend depth with the canvas... but on the other hand, the right way to do it would be to intialize depth to infinitely far (which I think is already happening), so this makes sense.